### ISSUE TITLE
mpv not working at all on Ubuntu 22.04 Wayland session

### ISSUE THREAD

**Issue by spirulin** [mpv-player/mpv#10152]
On X11 it works with zero problems.

On Wayland, it doesn't even open the player. Tried to reinstall again not working.

Provide following Information:
- mpv version: mpv 0.34.1
- Linux Distribution and Version: Ubuntu 22.04
- Source of the mpv binary: Ubuntu repos
- Window Manager and version: Gnome 42 under Wayland
- GPU driver and version: Latest NVIDIA drivers that come with 22.04.

To reproduce: Switch to Wayland if not, try to open MPV, or a file with it, nothing happens.

mpv 0.34.1 Copyright © 2000-2021 mpv/MPlayer/mplayer2 projects
 built on UNKNOWN
FFmpeg library versions:
   libavutil       56.70.100
   libavcodec      58.134.100
   libavformat     58.76.100
   libswscale      5.9.100
   libavfilter     7.110.100
   libswresample   3.9.100
FFmpeg version: 4.4.1-3ubuntu5

This is what I get when I try to play from the terminal something:

[vo/gpu/wayland] GNOME's wayland compositor lacks support for the idle inhibit protocol. This means the screen can blank during playback.
AO: [pulse] 48000Hz stereo 2ch float
VO: [gpu] 1920x1080 yuv420p
[destroyed object]: error 7: failed to import supplied dmabufs: Arguments are inconsistent (for example, a valid context requires buffers not supplied by a 
[vo/gpu/wayland] Error occurred on the display fd, closing
[mpvsucks.log](https://github.com/mpv-player/mpv/files/8604838/mpvsucks.log)

**Comment by CounterPillow** [comment]
Don't ignore the issue template, this is useless to us.

**Comment by spirulin** [comment]
Updated with more info. :)

**Comment by CounterPillow** [comment]
Thanks. Can you run mpv from the terminal with `mpv --log-file=mpvsucks.log a_video_here.mp4` and give us the log file it generates? That way we can hopefully see what goes wrong during context creation.

**Comment by Dudemanguy** [comment]
The chance of this working is probably low (why didn't they just keep xorg the default for nvidia?). If you're using EGLStreams, this is likely just broken. With NVIDIA's GBM support (I have no idea if the LTS has a new enough nvidia driver or not) this _may_ work however it requires a couple of commits that are in mpv's master branch and not in the 0.34.1 release.

tl;dr: Probably just use xorg unless you really want to use wayland for some reason.

**Comment by Dudemanguy** [comment]
Yeah so you're using 0.34.1 which definitely will not work on NVIDIA. Try building mpv from git master and seeing if that changes anything.

**Comment by thebombzen** [comment]
It appears that it worked. I'll re-open this issue if more info arrives suggesting the alternative.

**Comment by bogi788** [comment]
mpv 0.34.1 on Arch is working with nvidia (up to 510) on wayland if you're using gpu-api=vulkan and gpu-context=waylandvk

**Comment by Bananaman** [comment]
> mpv 0.34.1 on Arch is working with nvidia (up to 510) on wayland if you're using gpu-api=vulkan and gpu-context=waylandvk

Not true. Without those settings, you get a black/blue screen. If you add those settings, you get an image but there's extreme stuttering and freezing.

My guess is that the old mpv code (0.34.1) has a lot of old NVIDIA workarounds that no longer work now that NVIDIA runs native GBM Wayland instead of EGLStreams.

**Upgrading to git-master (latest code) fixes MPV on Wayland on NVIDIA, no config file needed.**

**Comment by Dudemanguy** [comment]
Just to clear up a few things here. There were never any nvidia-specific wayland workarounds. There were a couple of subtle implementation differences in NVIDIA's implementation that made egl on wayland broken on 0.34. These were fixed in a couple of commits that came later. Arch's 0.34.1 package actually cherrypicks a couple of commits so it should in theory work there (don't own a nvidia card myself so I do not know). Vulkan, in theory, should have worked fine since the day NVIDIA started supporting `VK_USE_PLATFORM_WAYLAND_KHR` but then again I don't own a NVIDIA card so I don't know.

>which probably causes mpv to use vsync instead of the wayland present protocol (implemented by the compositors).

mpv always uses the presentation time protocol on wayland if it supported by the compositor. This makes more of a clear difference if you use `display-resample`, but the vo takes it into account with the default audio sync mode as well.

**Comment by ideasman42** [comment]
Does anyone know which commit fixed this problem?
There is a similar [report for Blender](https://projects.blender.org/blender/blender/issues/104382).
